src/app.js

```diff
@@ -135,44 +134,46 @@ const App = Marionette.Application.extend({
 
     this.triggerMethod('before:start', options);
 
+    const opts = _.extend({}, options);
+    opts.state = this.getInitState(opts.state);
+
+    this.initState(opts);
+
     this._isRunning = true;
 
-    this.triggerStart(options);
+    this.triggerStart(opts);
 
     return this;
   },
 
   /**
-   * Triggers start event.
-   * Override to introduce async start
+   * Extends passed in state.
```

Comment:
@paulfalgout i guess this needs to be update as well?
<img width="504" alt="screen shot 2016-09-12 at 1 12 46 pm" src="https://cloud.githubusercontent.com/assets/7111256/18447068/a3e3f76c-78ea-11e6-905b-3074aa99290f.png">